File: jan10/jan10.java
public class SearchInsertPosition {
    public static int searchInsert(int[] nums, int target) {
        int left = 0;
        int right = nums.length;

        while (left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return left;
    }

    public static void main(String[] args) {
        int[] nums = {1, 3, 5, 6};
        int target1 = 5;
        int target2 = 2;

        System.out.println(searchInsert(nums, target1)); // Output: 2
        System.out.println(searchInsert(nums, target2)); // Output: 1
    }
}

File: jan10/jan10.js
function searchInsert(nums, target) {
    let left = 0;
    let right = nums.length;

    while (left < right) {
        const mid = Math.floor((left + right) / 2);
        if (nums[mid] === target) {
            return mid;
        } else if (nums[mid] < target) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }

    return left;
}

const nums = [1, 3, 5, 6];
const target1 = 5;
const target2 = 2;

console.log(searchInsert(nums, target1)); // Output: 2
console.log(searchInsert(nums, target2)); // Output: 1